lib/subject_selector.rb

```diff
@@ -3,47 +3,38 @@ class MissingParameter < StandardError; end
 
   attr_reader :user, :params, :workflow
 
-  def initialize(user, workflow, params, scope, session)
-    @user, @workflow, @params, @scope, @session = user, workflow, params, scope, session
+  def initialize(user, workflow, params, scope)
+    @user, @workflow, @params, @scope = user, workflow, params, scope
   end
 
   def queued_subjects
     raise workflow_id_error unless workflow
-    user_enqueued = UserSubjectQueue
-                    .find_by!(user: user.user, workflow_id: params[:workflow_id])
-    selected_subjects(user_enqueued.next_subjects(default_page_size))
-  end
+    raise group_id_error if needs_set_id? 
```

Comment:
@camallen do you mean something like the groups/random/subjects endpoint in Ouroboros? Science Gossip falls back to that if a group isn't chosen by the user. Should give back random subjects, without any regard to group ID. There's also the groups/subjects endpoint, which returns a set of subjects from a single random group. OWD might use that if no diary is chosen.

Sounds like this can be done in the client, anyway.